### PR TITLE
Add timezone to server schedule models. 

### PR DESCRIPTION
**Pull request by AY-Chen**
WIP for timezone, making the PR now to help @kdehaan see what i'm doing. Will fix and add some tests and then remove the WIP.  

Tests finished, removing WIP.

### CODE REVIEW

File: blinds/blinds_schedule.py [hunk 1]
```diff
@@ -330,6 +337,49 @@ def hasConflict( timeBlockList ):
...
+        tzRegex = r"GMT([+-])(\d{4})" # matches things like GMT-0600
+        gmtTimezoneBase = "Etc/GMT"
+        
+        matches = re.search( tzRegex, gmtStr)
+
+        if not matches:
+            raise InvalidTimeZoneStringException( "Invalid timezone string: " + gmtStr )
+
+        plusminus = matches.group(1)
+        offset = int( matches.group(2) ) / 100
```
Comment (ishchow) [inline blinds/blinds_schedule.py]: In the conditional below you check if offset < -14 but you only ever have the offset being positive. You have to use plusminus when creating offset.

File: requirements.txt [hunk 2]
```diff
@@ -10,6 +10,7 @@ pandas>=1.0.1
 pvlib==0.7.1
 smbus2==0.3.0
 Flask-Cors==3.0.8
+tzlocal>=2.0.0
```
Comment (ishchow) [inline requirements.txt]: Missing pytz here unless you're not using it. It was imported in another file.

File: tests/blinds/test_blindsschedule.py [hunk 3]
```diff
@@ -178,12 +181,57 @@ def test_scheduleConflictChecks( self ):
...
+        # Timezone with negative offset
+        tzString = "GMT-0600"
+        expectedTzName = "Etc/GMT-6"
+        parsedTz = BlindsSchedule.tzFromGmtString( tzString )
+        assert( parsedTz.zone == expectedTzName )
+
+    '''
+    Test for timezone string parsing with error. This tests highlights unsupported formats
+    '''    
+    def test_invalidTzParsing( self ):
```
Comment (ishchow) [inline tests/blinds/test_blindsschedule.py]: I don't really want you to change this test but FYI, if you end up having to duplicate code a lot in tests in python you can use the @pytest.mark.parametrize decorator (see: https://docs.pytest.org/en/latest/parametrize.html) where you can add test cases as an array of tuples, so it makes it trivial to implement many test cases with only a few test functions.

File: blinds/blinds_schedule.py [hunk 4]
```diff
@@ -330,6 +337,49 @@ def hasConflict( timeBlockList ):
...
+        
+        return timezone( gmtTimezoneBase + plusminus + str( int( offset ) ) )
+
+    '''
+    Converts a pytz timezone of the type Etc/GMT-X to a string formatted like "GMT-XXXX" 
+    '''
+    @staticmethod
+    def tzToGmtString( tz ):
+        tzNameRegex = r"Etc/GMT([+-])(\d+)"
+        matches = re.search( tzNameRegex, tz.zone)
```
Comment (ishchow) [inline blinds/blinds_schedule.py]: You forgot null check for matches, it causes your tests to fail.

File: blinds/blinds_schedule.py [hunk 5]
```diff
@@ -21,6 +21,9 @@
 from enum import Enum
 import json
 import datetime 
+from pytz import timezone
```
Comment (ishchow) [inline blinds/blinds_schedule.py]: Still using pytz? I can't really tell, too many things named timezone lol

File: requirements.txt (hunk 2 above)
Comment (AY-Chen) [reply requirements.txt]: pytz is included in python stdlib now, apparently. When i tried installing it, it told me it already was installed so i didn't inculde it.

Comment (ishchow) [reply requirements.txt]: oh i see, that's cool!

File: tests/blinds/test_blindsschedule.py (hunk 3 above)
Comment (AY-Chen) [reply tests/blinds/test_blindsschedule.py]: Good to know. Thanks :)

File: blinds/blinds_schedule.py (hunk 1 above)
Comment (AY-Chen) [reply blinds/blinds_schedule.py]: Fixed.

File: blinds/blinds_schedule.py (hunk 4 above)
Comment (AY-Chen) [reply blinds/blinds_schedule.py]: Fixed now.

**Pull request merged**